src/main/java/seedu/duke/parsermanager/ParseDeleteClient.java

```diff
@@ -50,19 +55,28 @@ public Command parseCommand() throws InvalidIndexException, MissingFlagException
             throw new NotIntegerException(MESSAGE_NOT_INTEGER);
         } catch (EmptyDetailException e) {
             throw new EmptyDetailException(MESSAGE_DELETE_CLIENT_WRONG_FORMAT);
+        } catch (ExtraFlagsException e) {
+            throw new ExtraFlagsException(MESSAGE_DELETE_CLIENT_WRONG_FORMAT);
         }
     }
 
     private ArrayList<String> processCommandDetails(String rawCommandDetail)
-            throws MissingFlagException, IncorrectFlagOrderException {
+            throws MissingFlagException, IncorrectFlagOrderException, ExtraFlagsException {
 
         String[] flags = DELETE_CLIENT_FLAGS;
         int[] flagIndexPositions = getFlagIndexPositions(rawCommandDetail, flags);
+        checkForExtraFlags(flagIndexPositions);
         checkForMissingFlags(flagIndexPositions);
         checkFlagsOrder(flagIndexPositions);
         return extractCommandDetails(rawCommandDetail, flags, flagIndexPositions);
     }
 
+    private void checkForExtraFlags(int[] flagIndexPositions) throws ExtraFlagsException {
+        if (flagIndexPositions[0] != CORRECT_FLAG_POSITION) {
```

Comment:
Perhaps you can consider replace '0' with another name?
Would it also be more readable if the entire boolean expression is replaced with a boolean variable with meaningful name?